### ISSUE TITLE
Noting Paper 50 - Revised Operating Model

### ISSUE THREAD

**Issue by JamesMBligh** [ConsumerDataStandardsAustralia/standards#50]
Looking ahead to how the API Working Stream is maintained and enhanced after the API standards is formally defined as version 1 a proposed governance model has been developed and reviewed internally by the Consumer Data Standards body.  This governance model is documented in the attached PDF.
[Noting Paper - Revised Operating Model.pdf](https://github.com/ConsumerDataStandardsAustralia/standards/files/2928623/Noting.Paper.-.Revised.Operating.Model.pdf)

Note that this does not represent a final decision on the model.  Feedback as to the feasibility and practical implications of this governance model is specifically being sought.

It is proposed that feedback on the initial model be open until March 15th.  The model will then be revised and republished for comment a second time for a longer period.

**Comment by NationalAustraliaBank** [comment]
Thank you for opening up feedback on the Revised Operating Model.

Overall, we support the types of decisions and criteria of the two change modes, and we'd like to suggest the introduction of an exception change mode. We also have some feedback and suggestions relating to change notifications and the proposed cadence / plan.

## Modes of change notification
Is there a preliminary step missing before defining the modes? Refer to subsequent sections for each change mode.

### Substantial Change Mode notifications
For the Substantial Change Mode, we would expect that there is a formal notification that a substantial change activity is formally noted to the public ahead of time. For example, typical events would be a new designation of an industry by Treasury or the Advisory Forum minutes outlining an upcoming period of substantial change occurring. It must be clear as to the mechanisms whereby an industry and/or the general public is advised of this event occurring.

### Steady State Change Mode notifications
For the Steady State Mode, again, what would be the preliminary steps that notifies the industry and the public that change will be initiated. As mentioned in the noting paper, the types of decisions are much more nuanced as it would typically not be aligned with a significant events, such as a new industry designation. We would like suggest that a regular cadence or calendar needs to be outlined so that organisations (including Data Holders and Data Recipients) can prepare for the engagement, as well as ensure appropriate funding exists to support the change.

This was considered as part of the feedback for the major and minor version of the endpoints (https://github.com/ConsumerDataStandardsAustralia/standards/issues/4), and that still remain relevant.

For example, a semi-annual cadence could be used:
-   Jan - Mar: Feedback period
-   Apr - Jun: Proposal review (steps 1-5)
-   Jul - Sep: Change window & feedback period
-   Oct - Dec: Proposal review
-   Jan - Mar: Change window & feedback period
-   A change cycle can be preserved (where appropriate) for a Substantial Change

## Rate of Steady State Change
We feel that the suggested operational cadence of six changes cycles a year needs to take into account lead time for organisations to make the changes as well account for other factors that were mentioned in Other Considerations.

Given that the intent of this noting paper in regards of the definition and formalisation of the standards only, how will the implementation and versioning operating model be integrated?

It is worth noting the observations of Open Banking in the UK that there was little change on end points and only a few new features introduced. A continuous and frequent change cadence has the potential to create significant time demand across participants, both Data Recipients and Data Holders, and as mentioned above, a less frequent but planned window could enable better planning and participation.

## Principle 2: Transparency
In line with our above feedback on change notification, we suggest that this principle be extended to provide a clear mechanism of communication to the industry and broader public that Data61 and ACCC are planning an upcoming change and the relative complexity of the change planned. The complexity can be used to determine the lead time involved.

## Exception Change Mode
There might be scenarios under which a change needs to be defined and implemented in an emergency / by-exception case and the severity of the change may mean that it is to be addressed outside of the Steady State mode.

How will impacted participants be notified of such changes? We appreciate that each exception and the response would be unique, and we suggest that some operational outlines and prioritisation (against other change modes) be provisioned in the Operating Model to cater for these scenarios as well as raise participants' awareness.

**Comment by JamesMBligh** [comment]
Thank you for the feedback @NationalAustraliaBank.  In response to the points raised:

- Communication of change events will be incorporated into the model.  Obviously, sometimes the initial communication of the change event (such as a new designated instrument) will occur outside the remit of the standards body but follow up comms is still important to incorporate in the model

- It is understood that the rate of change tabled may be faster than is required.  For the initial phase, however, it is anticipated that more change will occur before the end points stabilise.  Also, as more industries are added the domains that will be source of change will expand.  The note that the timing of the cadence would be open to change by the Lead of the API Working Group is a recognition of this.

- The changes to the standards do not imply the implementation times for the changes.  It is anticipated that part of the definition of a change to the standards will be a consultation on the likely times required for compliance as well as periods of maintenance and obsolescence for older versions.  There is no expectation that implementation must be immediate upon the acceptance of a change to the standards.  This was a key driver in the adoption of end point versioning.

- Consideration will be given to an exception change model.

-JB-

**Comment by JamesMBligh** [comment]
Due to a minimal number of responses being provided I will extend this item for another week.

-JB-

**Comment by anzbankau** [comment]
Hi James 
Some feedback/questions on the op model

1.	Steady state mode: what criteria are to be considered in establishing priority for change?  How will conflicting claims of priority amongst industry participants be handled?
2.	We see reference to removal and refactoring of properties for existing endpoints.  This implies breaking changes to the APIs, and later there is a reference to whether older versions of the same end points are to be made obsolete.  Is this covered in the versioning strategy?
3.	There is no information provided on whether there is any process to appeal the decision of the Chair.  Will appeals to the decision be heard?
4.	The 8-week cycle does not indicate the expectation of when the change is expected to be implemented.  Is this solely subject to Step 2 of the Steady State Decision Making Process?

**Comment by WestpacOpenBanking** [comment]
The Steady State process implies a rapid change cycle, which requires the different change lifecycles of the different participants to be accommodated. As such Westpac recommends that the steady state change process is limited to non-breaking changes only.

As such we recommended the Types of Changes catered for under Steady State Mode be described as:

The types of decisions that would be managed via this mode of operation would be those that do not introduce breaking changes, such as:
* The addition of a property or set of properties to an existing API endpoint
* The removal of a non-mandatory set of properties to an existing API endpoint
* The addition of new non-mandatory query parameters to an existing API endpoint
* The addition of new error responses to an existing API endpoint in a backwards compatible way
* A new end point allowing access to data that is already available in a new form
* A new end point that expands the capability of the overall API set but that is still aligned with the ACCC interpretation of the Designation Instrument (i.e. it does not expand the scope of the CDR regime)

**Comment by RiskGov** [comment]
Hi James - further recent communication outside this forum, some comments/questions below on the draft Noting Paper – Revised Operating Model.
For transparency to other participants, I am an independent consultant.

Dual Modes
•	Agree with proposed intent to operate dual modes - initial cadence to be influenced by first starters however large changes more frequently than quarterly sound unfeasible.  Suggest this reviewed once operational.
•	More broadly than Change, what is the intended due diligence process for entities and/or sectors willing to participate but not mandated in the first tranche implementation?
Steady State Mode
•	Taking a product perspective, what entity owns the API?
•	How are Compliance obligations (eg OAIC, ACCC) considered/communicated in proposed changes?
•	Is there a support service for active APIs – current and earlier versions?
•	How are issues with ‘production’ APIs reported and managed?
Decision making process
•	Who will decide whether to bundle/separate items for decision?
•	Are there weighted criteria for making the decision eg customer impact (#, segment), entity type etc?
•	How will the owner of the proposed change demonstrate risk assessment and compliance?
•	How are rejected submissions managed?
•	Is ‘Data Standards’ interchangeable with ‘API Standards’?
•	What is the approach to prioritise proposed changes?
•	Suggest prioritisation is by consensus
•	How will any interdependencies between proposed changes be considered?
Operational Cadence
Step 1
•	Suggest a template(?) approach to submissions will drive consistency and efficiency
•	What is the %/# of votes on Github required to reach consensus for any given change?
•	Suggest complexity of proposed change/s be considered in addition to volume in each cycle
•	How will new changes be considered with rejected (Backlog?) changes from an earlier cycle?
Step 2
•	Suggest clarifying language to explicitly address backwards compatibility
•	Suggest consideration of a standard period of time after implementing the proposed change for obsolescence of previous end point
•	Compliance (by definition) is not optional.  Suggest clarifying this point to address partial/full compliance and consider a standard period of time ie. Binding roadmap to compliance.
•	Suggest clarifying how compliance will be demonstrated and independently verified.
•	Is it proposed all rejected changes automatically form the Backlog?
•	Who will manage the Backlog – if this approach is proposed?
Step 3
•	What are the requirements for scope and extent of testing of proposed changes?
•	How test progress/results reporting shared per iteration/test cycle?
•	What are /production lodgement go-live criteria?
Step 4
•	Is the intent that this second round of consultation for rejected round 1 change proposals only?
•	Does the same consensus threshold apply to second round ‘refined’ proposals?
Step 5
•	What appeal/dispute process is proposed if the decision to reject/defer is not accepted?
•	Suggest clarifying if a Backlog will be maintained or if all rejected/deferred submissions must be resubmitted in full

Very important to clarify how the described process will integrate with other Agencies and CDR work streams.
Suggest communication and documentation should be mandatory for each release prior to implementation regardless of scale and complexity of change.  Need to standardise this requirement to ensure transparency and repeat-ability.
Substantial Change Mode
What is the proposed process and decision making criteria to secure endorsement for this approach?
Decision Making Process
Step 1
•	Suggest this include a documented and shared risk and compliance assessment
Driving Principles
Suggest inclusion of documented risk assessment at Principle #2 and #4 
End comments.

**Comment by commbankoss** [comment]
Commonwealth Bank do not recommend that there be a requirement for steady and substantial change modes. We suggest one operating model which determines decision making and operational cadence to preclude the issue of allocating and aligning change between processes. This will ensure that there is a clear and effective way that the community can engage with Data61. We have concerns that the current mode of change is creating confusion amongst stakeholders and is not providing the transparency that the community is seeking. 
 
We recommend a framework that focuses on community engagement and continuing the focus on consumer at the centre of the CDR. The principles which drive such frameworks are well articulated in the City of Sydney community engagement framework, being

- Informing, takes place when a decision has already been made or action is required
- Consulting, takes place when input is required, feedback or advice
- Active participation, collaborating with specific stakeholder groups

 
The desired outcome can be achieved through the application of Agile delivery practices – iterative evolution of requirements and design with on-going stakeholder engagement - aligned to these principles. We recommend adopting and embedding these principles alongside an Agile delivery framework as the basis of the operating model for the CDR Standards. This will drive a robust operating model that will help create transparency and collective participation across industries. 
 
Agile methodology is recommended for initiatives that endeavour to encompass stakeholder and team engagement[1].  
 
The adopted methodology must:
•	Create opportunities to continually refine and reprioritise a published backlog to align with community outcomes. 
•	Maximise the effort of the entire group by solving for the important issues in a time bound manner
•	Break down issues into manageable units
•	Remove the risks that traditionally occur when a continuous feedback circle is incomplete.
 
Traditional Agile is combination of people, process and mindset. There are key roles and ceremonies that need to be retained to maximise the results. We recommend embedding these ceremonies into the operating model so that decisions are transparent and the cadence is predictable (appendix 2). To empower the communities to easily track decisions and issues we recommend adapting a project board that can be accessed virtually (appendix 3). Lastly we want to acknowledge that a clear versioning strategy is imperative to work alongside the operating model to help track impending changes that the engineering community will need to support. 
![image](https://user-images.githubusercontent.com/43626361/62521150-6d3ce280-b872-11e9-8216-216f863b934e.png)

Commonwealth Bank believe that having two change modes of operational cadence would create inconstancy with the priorities and decisions made. By having one well-defined model that encompasses the agile framework will support the communities ability to influence and engage with CDR Standards change process. Once a decision has been made by the DSB chair it then should fall onto the community to decide if the change is either a bug fix, minor or major change. If the decision made is a breaking change to the standard this should be deferred until a major release. We would recommend that Data61 revise the versioning model so that endpoints, APIs and documentation can be versioned in a single model. Aligning to industry Standards of versioning and removing the dependency of complex versioning of independent endpoints.  
 
As an example of how these work together, 
 
•	DSB Chair decided to amend the definition of the pagination rules, this is a breaking change and is deferred to a major release as a feature and updates branch one with code changes.
•	Delivery team updates a typo in the documentation, this is not breaking change and is a bug fix. The updates to the master version are applied immediately.
•	Delivery team recommend addition of an enumerated featureType, this could be a breaking change depending on implementation. Community may decide to defer to a major release.
•	DSB wants to extend transaction API with new field. This is not a breaking change and can be committed to the master as a minor release change. 


![image](https://user-images.githubusercontent.com/43626361/62521222-90679200-b872-11e9-8f4c-24927553ce31.png)

### Appendix ###
 
#### 1.	Three Principles of Community Engagement ####
 
Reference: https://www.cityofsydney.nsw.gov.au/__data/assets/pdf_file/0010/291979/FINAL-Community-Engagement-Strategy-2017_DE5_web.pdf
 
#### 2.	Agile Roles and Ceremonies #####
 
**Roles**

- 	PO 
       - Data Standards Body Chair (Primary decision maker) 
       - Industry Stakeholders (Secondary influencers) 
       - Delivery Team 
       - API Lead
       - Engineering Lead 
       - Info Sec Lead 
       - CX Lead 
       - Scrum Master 
       - The role is to ensure that the team is supported in the Agile practices and process. 
       -Maintains healthy relationship dynamics between team, PO and others outside the team. 
	An effective scrum master can address distractions and keep the delivery team focused on outcomes. 
- We think this is an imperative role that will help facilitate the principles of community engagement. 

**Ceremonies**  
 
- Backlog Refinement 
       - The team review items on the backlog to ensure the backlog contains the appropriate items, that they are prioritised, and that the items at the top of the backlog are ready for delivery [3]. 
       - 1 Hour meeting occurring fortnightly. Needs to happen before the upcoming sprint. 
       - Open invite to stakeholders 
       - The final decision for top priorities is the Data Standards Body Chair  
       - Backlog board which can be visible from GitHub. 
       - To help facilitate this conversation we would recommend that a scrum master be present to facilitate conversion.
       - This session is optional for delivery teams. Can help inform but will only estimate effort in sprint planning. 
       - Mandatory for Data Standards Body Chair 
       - Video/Audio conference option  
 
- Sprint Planning 
       - The team determines amount of work which will be taken from the product backlog items and discusses their initial plan for completing those product backlog items [2].
       - Scrum master to facilitate conversation 
       - Each work item is assigned to a delivery team member 
       - These items will transition on a board from backlog to in progress. 
       - Tag issues on GitHub in the story item, helps with tractability. 
       - PO present but only commenting on priorities, not dictating estimation of work.  
       - Delivery team does estimations. 
       - 1 hour meeting occurring fortnightly. 
       - Other stakeholders optional
       - Video/Audio conference option  


- Daily Stand Ups 
       - Quick 15 minute meeting to give timely updates on the progression of the sprint goals. This should entail three standard questions to help facilitate outcomes.  What happened yesterday, what is planned for today and is there anything blocking or at risk to your work. 
       - Mandatory for delivery teams and scrum master 
       - Optional for other stakeholders and PO
       - Video/Audio conference option  

- Sprint Retrospectives
       - A retrospective is intended to reveal facts or feelings which have measurable effects on the team’s performance, and to construct ideas for improvement based on these observations [4]. 
       - Easiest form is to use three questions. What has worked well?  What was not a good experience? And what are areas for improvement?
       - Scrum Master accountable for improvements to the process 
       - Keep a register of any actions.
       - Mandatory for delivery teams and scrum master 
       - Optional for other stakeholders and PO
       - Video/Audio conference option  

- Sprint Review 
       - The Sprint Review takes place at the end of the Sprint and is designed to gather actionable feedback on what the Team has completed [5]. 
       - The delivery team walk through outcomes of the sprint. Answering questions about the increment.
       - The delivery team identify what successes or challenges occurred in the sprint. 
       - Escalate risks or blocks that might affect any work in future sprints. 
       - Once something is presented back to the audience the PO can declare the item completed or requiring more work. 
       - Actions of this meeting usually end on the backlog for prioritisation. 
       - Mandatory PO and stakeholders 
       - Optional for delivery team 
       - Video/Audio conference option  
 

 
 
#### 3.	Virtual Project Board #### 
 
Commonwealth Bank would recommend adopting a project board. As Data61 support GitHub to track issues we would recommend using a GitHub adaptable board. For setup please use this guide. We would recommend to have tracking lanes that translate with the operational cadence of the ceremonies. 
 
Example Project Board 
 

Backlog | In Progress | Review | Done
-- | -- | -- | --
·          List of issues that have been prioritised by   the community and PO.   ·          These can be created by anyone in the community   but priority will be managed by the Data Standards Body Chair. | ·          The current sprint issues in progress   ·          This can either be creation of completely new   decision proposals    ·          Amendments to the Standards    ·          The community are reviewing open issues. | ·          Items for Sprint review    ·          These are issues that need to be present back   to wider stakeholders | ·          Data Standards Chair has made a final decision      ·          To be done an issue has needed to have been   reviewed

**Comment by CDR-API-Stream** [comment]
Thanks for the feedback on this issue.  The API/InfoSec stream will be trialling an approach with an expectation that the trial will commence once the baseline version of the standards are published in response to the lock down rules.

More information on the trial process will be published shortly with the feedback provided above incorporated as much as possible.

**Comment by CDR-API-Stream** [comment]
The finalised options paper with feedback incorporated from consultation and from Advisory Committee review is attached below:
[Noting Paper - Standards Maintenance Trial.pdf](https://github.com/ConsumerDataStandardsAustralia/standards/files/3721524/Noting.Paper.-.Standards.Maintenance.Trial.pdf)

This attachment includes detail of the initial trial of the maintenance operating model.

The trial will be managed via the maintenance repository located at:
https://github.com/ConsumerDataStandardsAustralia/standards-maintenance/issues

Please refer to the README of this repository for more details.
